src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -621,7 +690,7 @@ private static string FormatDiagnostics(ImmutableArray<DiagnosticAnalyzer> analy
                 if (applicableAnalyzer != null)
                 {
                     var analyzerType = applicableAnalyzer.GetType();
-                    var rule = applicableAnalyzer.SupportedDiagnostics.Length == 1 ? string.Empty : $"{analyzerType.Name}.{diagnosticsId}";
+                    var rule = location != Location.None && location.IsInSource && applicableAnalyzer.SupportedDiagnostics.Length == 1 ? string.Empty : $"{analyzerType.Name}.{diagnosticsId}";
```

Comment:
seems weird that we are getting `default` or not-in-source locations here. Why is this? Is there a better place to filter them out?

Reply:
The testing framework supports validation of no-location diagnostics. This line is just a minor change to failure messaging; if there is more to do here it seems reasonable to leave it as a follow-up.